### PR TITLE
Document DRS endpoints (#1771)

### PR DESCRIPTION
**Pull request by jessebrennan**
According to Hannes and Trevor, documenting DOS is unnecessary.

### CODE REVIEW

File: lambdas/service/app.py [hunk 1]
```diff
@@ -194,6 +194,10 @@
         {
             'name': 'Manifests',
             'description': 'Complete listing of files matching a given filter in TSV and other formats'
+        },
+        {
+            'name': 'DRS',
+            'description': 'DRS compliant proxy for the underlying datastore'
```
Comment (hannes-ucsc) [inline lambdas/service/app.py:200]:
```suggestion
            'description': 'DRS-compliant proxy of the underlying repository'
```

File: lambdas/service/app.py [hunk 2]
```diff
@@ -2204,7 +2207,58 @@ def cart_export_send_to_collection_api(event, context):
...
-@app.route(drs.drs_http_object_path('{file_uuid}'), methods=['GET'], cors=True)
+drs_description = format_description('''
+    This is a partial implementation of the
+    [DRS 1.0.0 spec](https://ga4gh.github.io/data-repository-service-schemas/preview/release/drs-1.0.0/docs/).
+    Not all features are implemented. This endpoint acts as a DRS compatible
+    proxy for the underlying datastore.
+
+    Any errors encountered from the underlying datastore are forwarded on as
+    errors from this endpoint.
+''')
```
Comment (hannes-ucsc) [inline lambdas/service/app.py:2218]: Make this text consistent with the description changes I requested.

File: lambdas/service/app.py [hunk 3]
```diff
@@ -2204,7 +2207,58 @@ def cart_export_send_to_collection_api(event, context):
...
+            'description': format_description('''
+                A DRS object is returned. Either `access_id` will be included or
+                else both `access_url` and `type` are included.
+
+                If an `access_url` is returned, `type` will be `https` and
+                result is a signed URL to the object.
+
+                If an `access_id` is returned, use it in the next request.
+            '''),
+            'content': {
```
Comment (hannes-ucsc) [inline lambdas/service/app.py:2241]: Noah and I added a schema helper for this `content` - `application/json` structure.

File: lambdas/service/app.py [hunk 4]
```diff
@@ -2233,7 +2287,50 @@ def get_data_object(file_uuid):
...
+                    'description': 'Follow for redirection',
+                    'schema': {'type': 'string', 'format': 'url'},
+                }
+            }
+        }
+    },
+    'tags': ['DRS']
+})
 def get_data_object_access(file_uuid, access_id):
     """
```
Comment (hannes-ucsc) [inline lambdas/service/app.py:2335]: I think we should remove the docstring after making sure it is covered by the spec.

File: lambdas/service/app.py [hunk 5]
```diff
@@ -2233,7 +2287,50 @@ def get_data_object(file_uuid):
...
+
+        This method only needs to be called when using an `AccessMethod` that
+        contains an `access_id`
+    ''') + drs_description,
+    'parameters': [
+        params.path('file_uuid', str, description='UUID of the file to be fetched'),
+        params.path('access_id', str, description='Access ID returned from a previous request'),
+    ],
+    'responses': {
+        '301': {
```
Comment (hannes-ucsc) [inline lambdas/service/app.py:2304]: I'm confused. I thought the DRS standard doesn't have a place for 301s and 302s. Are we not implementing DRS correctly?

Comment (jessebrennan) [reply lambdas/service/app.py:2304]: You're right. The implementation is correct, but I screwed up on the documentation.

File: lambdas/service/app.py (hunk 3 above)
Comment (jessebrennan) [reply lambdas/service/app.py:2241]: I see this is still in https://github.com/DataBiosphere/azul/pull/1798 which isn't yet merged. Do you want me to chain on top of this branch?

File: lambdas/service/app.py (hunk 5 above)
Comment (jessebrennan) [reply lambdas/service/app.py:2304]: This was a pretty gaping mistake in the spec... I've looked over the code carefully now and I believe it is correct.

File: lambdas/service/app.py (hunk 3 above)
Comment (hannes-ucsc) [reply lambdas/service/app.py:2290]: The helper is already on the `develop` branch. Search for `application/json`

File: lambdas/service/app.py [hunk 6]
```diff
@@ -2253,7 +2256,58 @@ def cart_export_send_to_collection_api(event, context):
     }
 
 
-@app.route(drs.drs_http_object_path('{file_uuid}'), methods=['GET'], cors=True)
+drs_description = format_description('''
+    This is a partial implementation of the
+    [DRS 1.0.0 spec](https://ga4gh.github.io/data-repository-service-schemas/preview/release/drs-1.0.0/docs/).
+    Not all features are implemented. This endpoint acts as a DRS compatible
```
Comment (hannes-ucsc) [inline lambdas/service/app.py:2262]:
```suggestion
    Not all features are implemented. This endpoint acts as a DRS-compliant
```

File: lambdas/service/app.py [hunk 7]
```diff
@@ -2253,7 +2256,58 @@ def cart_export_send_to_collection_api(event, context):
...
+    proxy for the underlying repository.
```
Comment (hannes-ucsc) [inline lambdas/service/app.py:2263]:
```suggestion
    proxy for accessing files in the underlying repository.
```

File: lambdas/service/app.py (hunk 3 above)
Comment (jessebrennan) [reply lambdas/service/app.py:2290]: Thanks. I was only looking in `openapi/schema.py` before.

File: lambdas/service/app.py [hunk 8]
```diff
@@ -2283,15 +2335,46 @@ def get_data_object(file_uuid):
...
+    ''') + drs_description,
+    'parameters': [
+        params.path('file_uuid', str, description='UUID of the file to be fetched'),
+        params.path('access_id', str, description='Access ID returned from a previous request'),
+    ],
+    'responses': {
+        '202': {
+            'description': format_description('''
+                This response is issued if the object is not yet ready. Respect
+                the `Retry-After` header, then try again.
```
Comment (hannes-ucsc) [inline lambdas/service/app.py:2359]: I thought there is no such header. If this is our own deviation from the standard, we should explicit declare that fact.

Comment (jessebrennan) [reply lambdas/service/app.py:2359]: This is as per the DRS spec https://ga4gh.github.io/data-repository-service-schemas/preview/release/drs-1.0.0/docs/#_responses

**Pull request merged**